state/modelmigration.go

```diff
@@ -837,38 +837,26 @@ func (st *State) LatestMigration() (ModelMigration, error) {
 	return mig, nil
 }
 
-// CompletedMigrationForModel returns the most recent ModelMigration (if any)
-// for a model that has been removed from the state after
-// it reached the DONE phase and caused the model to be relocated.
-func (st *State) CompletedMigrationForModel() (ModelMigration, error) {
-	mig, phase, err := st.latestMigration(st.ModelUUID())
-	if err != nil {
-		return nil, err
-	}
-
-	// Return NotFound if the model still exists or the migration is not
-	// flagged as completed.
-	model, _ := st.Model()
-	if phase != migration.DONE || model != nil {
-		return nil, errors.NotFoundf("migration")
-	}
-
-	return mig, nil
+// CompletedMigration returns the most recent migration for this state's
+// model if it reached the DONE phase and caused the model to be relocated.
+func (st *State) CompletedMigration() (ModelMigration, error) {
+	mig, err := st.CompletedMigrationForModel(st.ModelUUID())
+	return mig, errors.Trace(err)
 }
 
-// CompletedMigration returns the most recent ModelMigration (if any)
-// for input model UUID that has been removed from the state after
-// it reached the DONE phase and caused the model to be relocated.
-func (st *State) CompletedMigration(modelUUID string) (ModelMigration, error) {
+// CompletedMigration returns the most recent migration for the
+// input model UUID if it reached the DONE phase and caused the model
+// to be relocated.
+func (st *State) CompletedMigrationForModel(modelUUID string) (ModelMigration, error) {
```

Comment:
Some of the churn is due to this exchange of method names that better indicate their behaviour.